pypit/arproc.py

```diff
@@ -930,6 +930,20 @@ def reduce_frame(slf, sciframe, scidx, fitsdict, det, standard=False):
         flex_dict = arwave.flexure_obj(slf, det)
         arqa.flexure(slf, det, flex_dict)
 
+    # Helio-centric
+    if settings.argflag['reduce']['calibrate']['refframe'] == 'heliocentric':
+        if settings.argflag['science']['extraction']['reuse'] == True:
+            msgs.warn("Heliocentric correction will not be applied if an extracted science frame exists, and is used")
+        msgs.work("Perform a full barycentric correction?")
+        #msgs.work("Include the facility to correct for gravitational redshifts and time delays (see Pulsar timing work)")
+        msgs.info("Performing a heliocentric correction")
+        # Load the header for the science frame
+        arwave.helio_corr(slf, det, fitsdict)
```

Comment:
If we allow this to also include a barycentric correction, perhaps we should rename the function to something more appropriate, e.g. `restframe`, `restframe_corr`, `wave_corr`, `corr_geomotion`, or something like that? Given how awful these suggestions are (!), I would be happy to go with whatever you think is appropriate, too.